### ISSUE TITLE
test_cli_pesel_generate_year_day[291-2207]

### ISSUE THREAD

**Issue by matee911** [jsporna/pypesel#4]
```python
✗ make test          
pytest -cov=pesel tests/
============================================================================================================== test session starts ===============================================================================================================
platform darwin -- Python 3.9.0, pytest-6.2.2, py-1.10.0, pluggy-0.13.1
rootdir: /Users/matee/Dev/3rd-party/pypesel/tests, configfile: ../ov=pesel
plugins: assume-2.4.2, cov-2.11.1
collected 835 items                                                                                                                                                                                                                              

tests/test_cli_pesel.py ...................................................................F.....................................................................................................................................          [ 23%]
tests/test_correct_pesel.py ............                                                                                                                                                                                                   [ 25%]
tests/test_generated_pesel.py ............................................................................................................................................................................................................ [ 49%]
.......................................................................................................................................................................................................................................... [ 77%]
..................................................................................................................................................................                                                                         [ 97%]
tests/test_incorrect_pesel.py ......................

==================================================================================================================== FAILURES ====================================================================================================================
___________________________________________________________________________________________________ test_cli_pesel_generate_year_day[291-2207] ___________________________________________________________________________________________________

    def cli():
        parser = argparse.ArgumentParser(prog="pesel", usage='%(prog)s')
    
        subparser = parser.add_subparsers(dest='command')
        generator = subparser.add_parser('generate', help='Generate PESEL number')
        generator.add_argument('--year', '-y', type=int, help='Year of birth',
                               default=argparse.SUPPRESS, dest='year')
        generator.add_argument('--month', '-m', type=int, help='Month of birth',
                               default=argparse.SUPPRESS, dest='month')
        generator.add_argument('--day', '-d', type=int, help='Day of birth',
                               default=argparse.SUPPRESS, dest='day')
        male_parser = generator.add_mutually_exclusive_group(required=False)
        male_parser.add_argument('--male', help='It is male person',
                                 default=argparse.SUPPRESS, action='store_true', dest='male')
        male_parser.add_argument('--female', help='It is female person',
                                 default=argparse.SUPPRESS, action='store_false', dest='male')
    
        validate = subparser.add_parser('validate', help='Validate PESEL number')
        validate.add_argument('pesel', metavar='pesel', type=str, help='PESEL number')
    
        args, _ = parser.parse_known_args()
        args = vars(args)
        command = args.pop('command')
    
        cmd = getattr(Pesel, command)
        try:
>           output = cmd(**args)

pesel/main.py:32: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

cls = <class 'pesel.pesel.Pesel'>, male = None, year = 2207, month = None, day = 29

    @classmethod
    def generate(cls, male: bool = None,
                 year: int = None, month: int = None, day: int = None):
        """Generate random PESEL number and create instance of class Pesel
    
        :param male: True for male, False for female
        :type male: bool
        :param year: The year of birth
        :type year: int
        :param month: The month of birth
        :type month: int
        :param day: The day of birth
        :type day: int
        :return: instance of class Pesel
        :rtype: pesel.Pesel
        """
        random.seed(time.time())
    
        gender = int(male) if male is not None else random.randint(0, 1)
    
        _year = year if year is not None else random.randint(Pesel.YEAR_MIN, Pesel.YEAR_MAX)
        if not Pesel.YEAR_MIN <= _year <= Pesel.YEAR_MAX:
            raise ValueError(f'Year should have value between {Pesel.YEAR_MIN} & {Pesel.YEAR_MAX}')
    
        _month = month if month is not None else random.randint(1, 12)
        if not 1 <= _month <= 12:
            raise ValueError('Month should have value between 1 and 12')
    
        max_day = calendar.monthrange(_year, _month)[1]
        if day and not month and day > max_day:
            if day == 31:
                _month = random.choice((1, 3, 5, 7, 8, 10, 12))
            if day == 30:
                _month = random.choice((1, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12))
    
        if day == 29 and not year and day > max_day:
            while True:
                _year = random.randint(Pesel.YEAR_MIN, Pesel.YEAR_MAX)
                if calendar.isleap(_year):
                    break
    
        max_day = calendar.monthrange(_year, _month)[1]
        _day = day if day is not None else random.randint(1, max_day)
        if not 1 <= _day <= max_day:
>           raise ValueError('Day should have value between 1 and {}'.format(max_day))
E           ValueError: Day should have value between 1 and 28

pesel/pesel.py:166: ValueError

During handling of the above exception, another exception occurred:

monkeypatch = <_pytest.monkeypatch.MonkeyPatch object at 0x103f05e50>, capsys = <_pytest.capture.CaptureFixture object at 0x103f055b0>, year = 2207, day = 29

    @pytest.mark.parametrize("year", [random.randrange(1800, 2299) for _ in range(5)])
    @pytest.mark.parametrize("day", [random.randrange(1, 31) for _ in range(5)])
    def test_cli_pesel_generate_year_day(monkeypatch, capsys, year, day):
        monkeypatch.setattr("sys.argv", ["pesel", "generate", "--year", str(year), "--day", str(day)])
>       cli()

tests/test_cli_pesel.py:63: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

    def cli():
        parser = argparse.ArgumentParser(prog="pesel", usage='%(prog)s')
    
        subparser = parser.add_subparsers(dest='command')
        generator = subparser.add_parser('generate', help='Generate PESEL number')
        generator.add_argument('--year', '-y', type=int, help='Year of birth',
                               default=argparse.SUPPRESS, dest='year')
        generator.add_argument('--month', '-m', type=int, help='Month of birth',
                               default=argparse.SUPPRESS, dest='month')
        generator.add_argument('--day', '-d', type=int, help='Day of birth',
                               default=argparse.SUPPRESS, dest='day')
        male_parser = generator.add_mutually_exclusive_group(required=False)
        male_parser.add_argument('--male', help='It is male person',
                                 default=argparse.SUPPRESS, action='store_true', dest='male')
        male_parser.add_argument('--female', help='It is female person',
                                 default=argparse.SUPPRESS, action='store_false', dest='male')
    
        validate = subparser.add_parser('validate', help='Validate PESEL number')
        validate.add_argument('pesel', metavar='pesel', type=str, help='PESEL number')
    
        args, _ = parser.parse_known_args()
        args = vars(args)
        command = args.pop('command')
    
        cmd = getattr(Pesel, command)
        try:
            output = cmd(**args)
        except ValueError as err:
            print(err, file=sys.stderr)
>           sys.exit(1)
E           SystemExit: 1

pesel/main.py:35: SystemExit
-------------------------------------------------------------------------------------------------------------- Captured stderr call --------------------------------------------------------------------------------------------------------------
Day should have value between 1 and 28
============================================================================================================ short test summary info =============================================================================================================
FAILED tests/test_cli_pesel.py::test_cli_pesel_generate_year_day[291-2207] - SystemExit: 1
========================================================================================================= 1 failed, 834 passed in 7.39s ==========================================================================================================
make: *** [test] Error 1
```

**Comment by jsporna** [comment]
OK, thx, I've found bug in tests :)

**Comment by jsporna** [comment]
fix is added in version 0.2.1
